front-end/src/views/function/components/AddFunc.vue

```diff
@@ -0,0 +1,242 @@
+        <el-input style="width: 95%"
+                  placeholder="请输入Input"
+                  v-model="form.inputs[index]"
+                  clearable>
+        </el-input>
+        <i style="float: right;font-size: 30px;width: 5%"
+           v-show="form.inputs.length!==1"
+           class="el-icon-circle-close"
+           @click="deleteInput(index)">
+        </i>
+      </el-form-item>
+      <el-button style="width: 100%;font-size: 20px;margin-top: 5px"
+                 size="mini"
+                 class="el-icon-circle-plus-outline"
+                 @click="addInput">
+      </el-button>
+      <el-form-item label="Output:"
+                    style="width: 100%"
+                    prop="output">
+        <el-input v-model="form.output"
+                  placeholder="请输入Output"
+                  clearable>
+        </el-input>
+      </el-form-item>
+      <el-form-item label="Log-topic:"
+                    style="width: 100%"
+                    prop="logTopic">
+        <el-input v-model="form.logTopic"
+                  placeholder="请输入Log-topic"
+                  clearable>
+        </el-input>
+      </el-form-item>
+      <el-form-item label="ClassName:"
+                    style="width: 66%"
+                    prop="className">
+        <el-input v-model="form.className"
+                  placeholder="请输入ClassName"
+                  clearable>
+        </el-input>
+      </el-form-item>
+      <el-form-item label="Runtime:"
+                    style="width: 30%"
+                    prop="runtime">
+        <el-select v-model="form.runtime">
+          <el-option label="JAVA"
+                     value="JAVA">
+          </el-option>
+          <el-option label="GO"
+                     value="GO">
+          </el-option>
+        </el-select>
+      </el-form-item>
+      <el-form-item class="upload"
+                    label="File:"
+                    prop="file"
+                    style="width: 100%">
+        <el-upload
+          :limit="1"
+          drag
+          ref="file"
+          accept=".jar"
+          action="0.0//不写会报错，神奇"
+          :auto-upload="false"
+          :on-change="getFile">
+          <i class="el-icon-upload"></i>
+          <div class="el-upload__text">将文件拖到此处，或点击上传</div>
+          <div class="el-upload__tip">目前只支持jar文件上传</div>
+        </el-upload>
+      </el-form-item>
+      <div
+        :style="{
+          position: 'absolute',
+          right: 0,
+          bottom: 0,
+          width: '100%',
+          borderTop: '1px solid #e9e9e9',
+          padding: '10px 16px',
+          background: '#fff',
+          textAlign: 'right',
+          zIndex: 1
+        }">
+        <el-button type="primary" @click="onSubmit">立即创建</el-button>
+        <el-button @click="resetField('addForm')">重置</el-button>
+      </div>
+    </el-form>
+  </el-drawer>
+</template>
+
+<script>
+import { create } from '@/api/func'
+export default {
+  props:{
+    functionList: {
```

Comment:
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/FunctionStream/function-stream/pullRequest?prid=7834759)
```suggestion
      functionList: {
```